tests/Controller/CartApiTest.php

```diff
@@ -0,0 +1,546 @@
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_to_get_carts_list()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/cart.yml');
+
+        $this->client->request('GET', '/api/v1/carts/', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'cart/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_list_carts_in_state_different_than_cart()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/order.yml');
+
+        $this->client->request('GET', '/api/v1/carts/', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'cart/empty_index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_denies_carts_deletion_for_non_authenticated_user()
+    {
+        $this->client->request('DELETE', '/api/v1/carts/-1');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_not_found_response_when_trying_to_delete_cart_which_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('DELETE', '/api/v1/carts/-1', [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_to_delete_cart()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $carts = $this->loadFixturesFromFile('resources/cart.yml');
+
+        /** @var OrderItemInterface $cart */
+        $cart = $carts['order_001'];
+        $url = '/api/v1/carts/'.$cart->getId();
+
+        $this->client->request('DELETE', $url, [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, Response::HTTP_NO_CONTENT);
+
+        $this->client->request('GET', $url, [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_delete_orders_in_state_different_than_cart()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $carts = $this->loadFixturesFromFile('resources/order.yml');
```

Comment:
Since we're defining it as a non-cart object (as stated in the test's title), should the variables still be named cart?